core/remote/src/main/kotlin/team/noweekend/core/remote/provider/TokenProvider.kt

```diff
@@ -0,0 +1,10 @@
+package team.noweekend.core.remote.provider
+
+import kotlinx.coroutines.flow.Flow
+
+interface TokenProvider {
+    val accessTokenFlow: Flow<String>
+    val refreshTokenFlow: Flow<String>
```

Comment:
```suggestion
    val tokenFlow: Flow<Token>
    
    // :core:model
    data class Token(
        val accessToken: String,
        val refreshToken: String,
    )
```
개인적으로 요런 AuthToken을 관리하는 flow를 두개를 만들어서 각각 관리하는 것 보다는 data class하나 만들어서 하나로 관리하는게 더 좋을 것 같다는 생각이 있는데 어때?
물론 전제조건은 우리가 refreshToken을 쓰는 경우!

Reply:
훔. flow를 coremodel에서 가지고 있는 것이 조금 걱정되긴 하는데요.

토큰 값은 왠만하면 데이터 모듈이나 remote쪽 에서만 알고 있음 될거 같아서 다 접근할 수 있는 core model까지 빼는 거는 조금 의문이 들긴행.

하지만 remote모듈에서만 데이터 클래스로 묶는다면 그거는 괜찮은거 같기도 하고!

Reply:
> 훔. flow를 coremodel에서 가지고 있는 것이 조금 걱정되긴 하는데요.

Flow를 도메인 모델 단에서 들고있는 의도는 아니긴해! Token 모델만 model에서 유지하는 형식?

> 토큰 값은 왠만하면 데이터 모듈이나 remote쪽 에서만 알고 있음 될거 같아서 다 접근할 수 있는 core model까지 빼는 거는 조금 의문이 들긴행.

아직 구체적인 인증방식이 정해져있지 않다보니까 어떨지는 모르겠는데, UI단에서도 token을 바라봐야하는 경우도 있는 것 같아
예를들면, 사용자가 로그인 되어있을 때(로컬에 토큰정보가 있을 때) 로그아웃되어있을 때 (로컬에 토큰정보가 없을 때;empty인 경우)에 호출하는 API가 다른 경우?도 해당될 수 있을듯?
``` kotlin
val getLocalTokenUseCase = GetLocalTokenUseCase()
if (getLocalTokenUseCase.accessToken.isEmpty()) {
    // API 1 호출
} else {
    // API 2 호출
}
```

조금 애매한건 이게 remote에서부터 내려주는거다보니까 UI단에서 사용하게끔 하려면 DomainModel에 Token 모델을 하나 두고 사용하는 쪽에서 다 매퍼 만들어서 써줘야 할텐데, 조금 짜치는 상황일 수도 있어서.. 그래서 이럴때는 직접적으로 remote랑 feature에서도 core:model을 볼수있게 하는 의존성을 이 케이스에서만 허용할수도 있을 것 같엉
요건 논의가 필요할지도?
cc. @014967

Reply:
@JaesungLeee 훔. UI단에서 토큰을 직접적으로 바라보는 일은 사실 없을거 같다고 생각이 들긴 하는게, 토큰 비교는 repository 단 에서 비교하고 UI 단에서 boolean으로 판별할 수 있을거 같아서.

ui단에서는 사실 바라볼필요가 없을 거 같은데 직접 바라보는게 더 나을까??

Reply:
feature module에서 token model 직접적으로는 안볼꺼같엉
위의 예시를 return을 boolean으로 하면 될꺼같넹 @SEO-J17 형 말대로

Reply:
![image](https://github.com/user-attachments/assets/910b3a52-71b4-4725-9cd6-9890bf737ae6)